### PR TITLE
NP-2474: Service returns DTOs 

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/TriggerHandler.java [hunk 1]
```diff
@@ -98,10 +96,6 @@ private static DynamoDBCustomerService defaultCustomerService() {
             ENVIRONMENT);
     }
 
-    @JacocoGenerated
```
Comment (axthosarouris) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/TriggerHandler.java:101]: CustomerMapper is now a utility class. Its name will change on subsequent PR

File: customer-commons/src/main/java/no/unit/nva/customer/ObjectMapperConfig.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 public final class ObjectMapperConfig {
 
-    public static final ObjectMapper objectMapper = JsonUtils.objectMapper;
+    public static final ObjectMapper objectMapper = JsonUtils.objectMapperWithEmpty;
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/ObjectMapperConfig.java:8]: Necessary for ensuring that all serializations (including tests follow the pattern of the handler). Probably this class should be removed and we should rely on the default handler serializer. (But not in this PR)

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java [hunk 3]
```diff
@@ -30,7 +30,11 @@ public CustomerList(List<CustomerDto> customers) {
      * @return customerList
      */
     public static CustomerList of(List<CustomerDto> customers) {
-        return new CustomerList(customers);
+        List<CustomerDto> customersWithoutContext = customers
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java:33]: Moved functionality of CustomerMapper here. This creates a "null" context for each entry in the list, but this is a bug that will be fixed in another PR. In any case, the fix is now local to this class.

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java [hunk 4]
```diff
@@ -1,28 +1,23 @@
...
+import java.util.Objects;
 import java.util.Optional;
 import java.util.UUID;
-import java.util.stream.Collectors;
 import no.unit.nva.customer.model.CustomerDto.Builder;
 import nva.commons.core.Environment;
-import nva.commons.core.JsonUtils;
 
-public class CustomerMapper {
+public final class CustomerMapper {
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java:10]: This class has the functionality of just adding context to the DTOs. It will be renamed in later PR

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java [hunk 5]
```diff
@@ -36,55 +31,13 @@ public static URI toId(UUID identifier) {
...
-    /**
-     * Map from Customer from Dto to Db version.
-     *
-     * @param customerDto customerDto
-     * @return customerDb
-     */
-    public CustomerDb toCustomerDb(CustomerDto customerDto) {
-        CustomerDb customer = objectMapper.convertValue(customerDto, CustomerDb.class);
-        return customer;
+    public static CustomerDto removeContext(CustomerDto customerDto) {
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java:34]: Method for preserving the existing functionality. It will be removed once it is not used by CustomerList.

File: customer/src/main/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandler.java [hunk 6]
```diff
@@ -20,15 +20,10 @@
...
-import java.net.URI;
-import java.util.List;
-
 public class GetCustomerByOrgNumberHandler extends ApiGatewayHandler<Void, CustomerIdentifiers> {
 
-    public static final String ID_NAMESPACE_ENV = "ID_NAMESPACE";
-    public static final String ORG_NUMBER = "orgNumber";
 
-    private final CustomerMapper customerMapper;
+    public static final String ORG_NUMBER = "orgNumber";
```
Comment (axthosarouris) [inline customer/src/main/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandler.java:26]: removed CustomerMapper

File: customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java [hunk 7]
```diff
@@ -90,8 +61,22 @@ protected UUID getIdentifier(RequestInfo requestInfo) throws ApiGatewayException
...
+        return customerService.getCustomer(getIdentifier(requestInfo));
+    }
+
     @Override
     protected Integer getSuccessStatusCode(Void input, CustomerDto output) {
         return SC_OK;
     }
+
+    @JacocoGenerated
+    private static DynamoDBCustomerService defaultCustomerService() {
```
Comment (axthosarouris) [inline customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java:76]: defaultCustomerService method is duplicated in many classes, but now it can be defined once and re-used, in another PR.

File: customer/src/main/java/no/unit/nva/customer/update/UpdateCustomerHandler.java [hunk 8]
```diff
@@ -18,65 +18,31 @@
...
-            AmazonDynamoDBClientBuilder.defaultClient(),
-            ObjectMapperConfig.objectMapper,
-            new Environment());
-    }
-
-    @JacocoGenerated
-    private static CustomerMapper defaultCustomerMapper() {
-        String namespace = new Environment().readEnv(ID_NAMESPACE_ENV);
-        return new CustomerMapper(namespace);
+        this(defaultCustomerService(), new Environment());
```
Comment (axthosarouris) [inline customer/src/main/java/no/unit/nva/customer/update/UpdateCustomerHandler.java:32]: All handlers changed in very similar ways, due to their common structure. The changes are almost identical in every handler

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandlerTest.java [hunk 9]
```diff
@@ -34,14 +32,13 @@
 public class GetCustomerByOrgNumberHandlerTest {
 
     public static final String WILDCARD = "*";
-    public static final String SAMPLE_NAMESPACE = "http://example.org/customer";
     public static final String REQUEST_ID = "requestId";
     public static final String SAMPLE_ORG_NUMBER = "123";
     public static final String EXPECTED_ERROR_MESSAGE = "Missing from pathParameters: orgNumber";
-    public static final String SAMPLE_CRISTIN_ID = "http://cristin.id";
+    public static final String SAMPLE_CRISTIN_ID = "https://cristin.id";
```
Comment (axthosarouris) [inline customer/src/test/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandlerTest.java:38]: IDEA was complaining :)

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandlerTest.java [hunk 10]
```diff
@@ -70,9 +66,8 @@ public void getCustomerByOrgNumberReturnsCustomerWhenInputIsExistingCustomerOrgN
             .withFeideOrganizationId(SAMPLE_ORG_NUMBER)
             .withCristinId(SAMPLE_CRISTIN_ID)
             .build();
-        when(customerServiceMock.getCustomerByOrgNumber(SAMPLE_ORG_NUMBER)).thenReturn(customerDb);
-
-        CustomerDto customerDto = customerMapper.toCustomerDto(customerDb);
+        CustomerDto customerDto = customerDb.toCustomerDto();
```
Comment (axthosarouris) [inline customer/src/test/java/no/unit/nva/customer/get/GetCustomerByOrgNumberHandlerTest.java:69]: removed CustomerMapper

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java [hunk 11]
```diff
@@ -57,13 +56,11 @@
...
      */
     @BeforeEach
-    @SuppressWarnings("unchecked")
     public void setUp() {
         customerServiceMock = mock(CustomerService.class);
         environmentMock = mock(Environment.class);
-        customerMapper = new CustomerMapper(SAMPLE_NAMESPACE);
         when(environmentMock.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn(WILDCARD);
-        handler = new GetCustomerHandler(customerServiceMock, customerMapper, environmentMock);
+        handler = new GetCustomerHandler(customerServiceMock, environmentMock);
```
Comment (axthosarouris) [inline customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java:63]: removed CustomerMapper

**Review by janbj — APPROVED** [review]
